pkg/cmd/secret/delete/delete.go

```diff
@@ -88,6 +99,14 @@ func removeRun(opts *DeleteOptions) error {
 		return err
 	}
 
+	var baseRepo ghrepo.Interface
+	if secretEntity == shared.Repository || secretEntity == shared.Environment {
+		baseRepo, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
I assume this was moved higher up so any ambiguity errors are raised earlier before the other ways this can error.